### PR TITLE
Add bearer token authorization to the bundle route

### PR DESCRIPTION
**Pull request by garryod**
Adds bearer token authorization to the `/bundle.tar.gz` endpoint. The required token can be configured via the CLI.
This should be replaced with an OAuth2 compliant proxy once we have support for device flow in our IdP. In the meantime this is no worse than the `ispyb_ro` account which is used for the underlying database access

### CODE REVIEW

**Review by DiamondJoseph — COMMENTED** [review]
Agree on the necessity on a conceptual level; and passing appropriate credentials out of band for now, if @tpoliaw is able to approve on an implementation level

I see see the CLI argument being added, being read, being used, is this currently for all endpoints?

**Comment by garryod** [comment]
> I see see the CLI argument being added, being read, being used, is this currently for all endpoints?

Just for the `/bundle.tar.gz` endpoint (though this is also technically all endpoints)

File: bundler/src/require_bearer.rs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+///
+/// Requests which do not have a valid token are sent a 401 Unauthorized response
+#[derive(Clone)]
+pub struct RequireBearerLayer {
+    /// The required token value
+    required_token: Option<String>,
+}
+
+impl RequireBearerLayer {
+    /// Creates the [`tower::Service`] with a given required token
```
Comment (tpoliaw) [inline bundler/src/require_bearer.rs:24]: Service -> Layer?

**Review by tpoliaw — APPROVED** [review]
I don't know about all the k8s yaml but the middleware looks alright as a stop gap until the oauth stuff works.

**Pull request merged**